### ISSUE TITLE
Connection failed with Timeout

### ISSUE THREAD

**Issue by fiberwings** [iancmcc/ouimeaux#48]
Using the latest version I still had connections problems that were not being taken care by the new retry code:

```
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/gevent/greenlet.py", line 327, in run
    result = self._run(*self.args, **self.kwargs)
  File "/usr/local/lib/python2.7/dist-packages/pysignals/dispatcher.py", line 197, in send
    response = receiver(signal=self, sender=sender, **named)
  File "/usr/local/lib/python2.7/dist-packages/ouimeaux/environment.py", line 137, in _found_device
    device = klass(headers['location'])
  File "/usr/local/lib/python2.7/dist-packages/ouimeaux/device/__init__.py", line 28, in __init__
    service = Service(svc, base_url)
  File "/usr/local/lib/python2.7/dist-packages/ouimeaux/device/api/service.py", line 69, in __init__
    xml = requests_get(url)
  File "/usr/local/lib/python2.7/dist-packages/ouimeaux/utils.py", line 80, in inner
    return f(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/api.py", line 55, in get
    return request('get', url, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/api.py", line 44, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 456, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 559, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/adapters.py", line 384, in send
    raise Timeout(e, request=request)
Timeout: (<requests.packages.urllib3.connectionpool.HTTPConnectionPool object at 0x1bca330>, 'Connection to 192.168.0.5 timed out. (connect timeout=1)')
<Greenlet at 0x1b752b0: <bound method Signal.send of <pysignals.dispatcher.Signal object at 0x1b3ff50>>(<ouimeaux.discovery.UPnP object at 0x1b5a590>, headers={'opt': '"http://schemas.upnp.org/upnp/1/0/"; ns=0, address=('192.168.0.5', 3165))> failed with Timeout
```

This is because the Timeout exception was not handled. Fixed with a:

```
-            except requests.ConnectionError:
+            except (requests.ConnectionError, requests.Timeout) as e:
```

**Comment by logjames** [comment]
This change or something recent breaks maker support…I’ll try to figure it out this weekend.

**Comment by iancmcc** [comment]
Uh oh, I just released 0.7.9! Let me know if you get a chance to get anywhere with this and I'll cut a new release.
